instruments/thorlabs/thorlabsapt.py

```diff
@@ -1337,6 +1337,131 @@ def backlash_correction(self, pos):
+                Home Velocity,
+                Offset distance
+            :rtype: Tuple[int, int, u.Quantity, u.Quantity]
+
+
+            Example:
+                >>> import instruments as ik
+                >>> import instruments.units as u
+
+                >>> # load the controller, a KDC101 cube
+                >>> kdc = ik.thorlabs.APTMotorController.open_serial("/dev/ttyUSB0", baud=115200)
+                >>> # assign a channel to `ch`
+                >>> ch = kdc.channel[0]
+                >>> ch.motor_model = 'PRM1-Z8'  # select rotation stage
+
+                >>> # set offset distance to 4 degrees, leave other values
+                >>> ch.home_parameters = None, None, None, 4 * u.deg
+                >>> ch.home_parameters  # read it back
+                (2, 1, <Quantity(9.99, 'degree / second')>, <Quantity(3.99, 'degree')>)
+            """
+            pkt = _packets.ThorLabsPacket(
+                message_id=_cmds.ThorLabsCommands.MOT_REQ_HOMEPARAMS,
+                param1=self._idx_chan,
+                param2=0x00,
+                dest=self._apt.destination,
+                source=0x01,
+                data=None,
+            )
+            response = self._apt.querypacket(
+                pkt,
+                expect=_cmds.ThorLabsCommands.MOT_GET_HOMEPARAMS,
+                expect_data_len=14,
+            )
+            # chan, home_dir, limit_switch, velocity, ,offset_dist
+            _, home_dir, lim_sw, vel, offset = struct.unpack("<HHHll", response.data)
+            return (
+                int(home_dir),
+                int(lim_sw),
+                u.Quantity(vel) / self.scale_factors[1],
+                u.Quantity(offset, "counts") / self.scale_factors[0],
+            )
+
+        @home_parameters.setter
+        def home_parameters(self, values):
+            values = list(values)
+            if len(values) != 4:
+                raise ValueError(
+                    "Home parameters muust be set with four values: "
+                    "Home direction, limit switch settings, velocity, and offset. "
+                    "For settings you want to leave untouched, pass `None`."
+                )
+
+            # replace values that are `None`
+            if None in values:
+                set_params = self.home_parameters
+                for it in range(len(values)):
+                    if values[it] is None:
+                        values[it] = set_params[it]
+
+            home_dir, lim_sw, velocity, offset = values
+            # check validity of unitful values
+            if not isinstance(velocity, u.Quantity):
+                velocity = int(velocity)
+            else:
+                # Ensure velocity is dimensionless
+                req_units = (1 / self.scale_factors[1]).units
+                try:
+                    velocity = int(
+                        (velocity.to(req_units) * self.scale_factors[1]).magnitude
```

Comment:
I would have liked to include this in a nicer fashion and simply divide by the scaling factor. However, somehow `pint` seems to not notice dimensionless units when doing math, e.g.:

```python
>>> a = 1*u.m
>>> b = 1*u.cm
>>> print(a/b)
1.0 meter / centimeter
```

If you do the same with identical units, it comes out as `dimensionless`. Need to check more, might be a `pint` issue? The workaround here is functional

Reply:
Turns out, this can actually be done in `pint` and I should have just read the manual... Solution is:

```python
c=a/b
c.ito_reduced_units()
```
This makes it dimensionless.

Reply:
Yeah I was going to mention today that these libraries usually have a function that needs to be called to actually reduce the units to base SI